packages/global-styles/src/components/color-palette-variations/index.tsx

```diff
@@ -61,11 +62,19 @@ const ColorPaletteVariation = ( {
 				} ) as string
 			}
 		>
-			<div className="global-styles-variation__item-preview">
-				<GlobalStylesContext.Provider value={ context }>
-					<ColorPaletteVariationPreview title={ colorPaletteVariation.title } />
-				</GlobalStylesContext.Provider>
-			</div>
+			<InView triggerOnce>
```

Comment:
Avoid rendering all variations at once since it hurts the performance and makes the animation slow